tests/E2E/ClientTest.php

```diff
@@ -24,6 +25,24 @@
 
 class ClientTest extends BaseTestCase
 {
+    public function testUserAgent()
+    {
+        $client = $this->getUnboundClient();
+
+        $property = (new \ReflectionClass(Client::class))->getProperty('userAgentGenerator');
+        $property->setAccessible(true);
+        $generator = $property->getValue($client);
+
+        // PHP doesn't support the "g" modifier
+        // so we can't use PHPUnit's assertRegExp method
+        // and we need to rely on preg_match_all
+        // which returns "false" if no matches are found,
+        // or a number otherwise
+        $result = \preg_match_all('/(app|sdk|platform|integration|os) \S+(\/\d+.\d+.\d+(-[\w\d-]+)?)?;/im', $generator->getUserAgent());
+        $this->assertInternalType('int', $result);
+        $this->assertGreaterThan(0, $result);
```

Comment:
May be check greater than 2 or 3 since it should always have a SDK version + os + platform?

Otherwise: LGTM